### PR TITLE
[BOLT][DWARF][NFC] Add split-dwarf5 test with multiple CUs

### PR DESCRIPTION
**Pull request by sayhaan**
Adds a split-dwarf test for DWARF5 with multiple CUs.

### CODE REVIEW

File: bolt/test/X86/Inputs/dwarf5-df-input-lowpc-ranges-other.s [hunk 1]
```diff
@@ -0,0 +1,453 @@
...
+#   return val;
+# }
+# __attribute__((always_inline))
+# int doStuffOther2(int val) {
+#   int foo = 3;
+#   return val + foo;
+# }
+#
+#
+# int mainOther(int argc, const char** argv) {
```
Comment (ayermolo) [inline bolt/test/X86/Inputs/dwarf5-df-input-lowpc-ranges-other.s:15]: Are you sure mainOther was part of your cpp?

File: bolt/test/X86/Inputs/dwarf5-df-input-lowpc-ranges-other.s [hunk 2]
```diff
@@ -0,0 +1,453 @@
+# clang++ -fbasic-block-sections=all -ffunction-sections -g2 -gdwarf-5 -gsplit-dwarf
```
Comment (ayermolo) [inline bolt/test/X86/Inputs/dwarf5-df-input-lowpc-ranges-other.s:1]: Was it generated with: -fdebug-compilation-dir

Comment (sayhaan) [reply bolt/test/X86/Inputs/dwarf5-df-input-lowpc-ranges-other.s:1]: Yes, forgot to change the comment at the top. Will fix.

File: bolt/test/X86/Inputs/dwarf5-df-input-lowpc-ranges-other.s (hunk 1 above)
Comment (sayhaan) [reply bolt/test/X86/Inputs/dwarf5-df-input-lowpc-ranges-other.s:15]: Yes, this one wasn't working until I regenerated it.

File: bolt/test/X86/dwarf5-df-input-lowpc-ranges-cus.test [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+; RUN: -split-dwarf-file=main.dwo -o main.o
+; RUN: llvm-mc -dwarf-version=5 -filetype=obj -triple x86_64-unknown-linux %p/Inputs/dwarf5-df-input-lowpc-ranges-other.s \
+; RUN: -split-dwarf-file=mainOther.dwo -o other.o
+; RUN: %clang %cflags -gdwarf-4 -gsplit-dwarf=split main.o other.o -o main.exe
+; RUN: llvm-bolt main.exe -o main.exe.bolt --update-debug-sections
+; RUN: llvm-dwarfdump --show-form --verbose --debug-rnglists main.exe.bolt &> %t/foo.txt
+; RUN: llvm-dwarfdump --show-form --verbose --debug-addr main.exe.bolt >> %t/foo.txt
+; RUN: llvm-dwarfdump --show-form --verbose --debug-info main.exe.bolt >> %t/foo.txt
+; RUN: cat %t/foo.txt | FileCheck -check-prefix=BOLT %s
+; RUN: llvm-dwarfdump --show-form --verbose --debug-info main.dwo.dwo mainOther.dwo &> %t/mainddwodwo.txt
```
Comment (ayermolo) [inline bolt/test/X86/dwarf5-df-input-lowpc-ranges-cus.test:14]: mainOther.dwo is pre-bolt....

File: bolt/test/X86/Inputs/dwarf5-df-input-lowpc-ranges-other.s (hunk 1 above)
Comment (ayermolo) [reply bolt/test/X86/Inputs/dwarf5-df-input-lowpc-ranges-other.s:15]: I don't see mainOther symbol in your assembly....

File: bolt/test/X86/Inputs/dwarf5-df-input-lowpc-ranges-other.s [hunk 4]
```diff
@@ -0,0 +1,753 @@
+# clang++ -fbasic-block-sections=all -ffunction-sections -g2 -gdwarf-5 -gsplit-dwarf -fdebug-compilation-dir='.'
```
Comment (ayermolo) [inline bolt/test/X86/Inputs/dwarf5-df-input-lowpc-ranges-other.s:1]: double hashes

**Review by ayermolo — APPROVED** [review]
small nit, otherwise LGTM

**Comment by ayermolo** [comment]
image
can you cleanup the branches

File: bolt/test/X86/dwarf5-df-input-lowpc-ranges-cus.test [hunk 5]
```diff
@@ -0,0 +1,87 @@
+; RUN: rm -rf %t
+; RUN: mkdir %t
+; RUN: cd %t
+; RUN: llvm-mc -dwarf-version=5 -filetype=obj -triple x86_64-unknown-linux %p/Inputs/dwarf5-df-input-lowpc-ranges-main.s \
+; RUN: -split-dwarf-file=main.dwo -o main.o
+; RUN: llvm-mc -dwarf-version=5 -filetype=obj -triple x86_64-unknown-linux %p/Inputs/dwarf5-df-input-lowpc-ranges-other.s \
+; RUN: -split-dwarf-file=mainOther.dwo -o other.o
+; RUN: %clang %cflags -gdwarf-4 -gsplit-dwarf=split main.o other.o -o main.exe
```
Comment (ayermolo) [inline bolt/test/X86/dwarf5-df-input-lowpc-ranges-cus.test:8]: -gdwarf-4 -gsplit-dwarf=split is not necessary as we discussed

**Review by ayermolo — APPROVED** [review]

**Pull request merged**